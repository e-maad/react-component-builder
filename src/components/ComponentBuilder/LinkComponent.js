import React from 'react';
import { connect } from 'react-redux';

const LinkComponent = ({ text, url, id, connectedModal, dispatch }) => {
    
    const clickHandler = () => {
        if (connectedModal) {
            dispatch({
                payload: {
                    [connectedModal]: true,
                    [`${connectedModal}Trigger`]: id
                },
                type: ''
            })
        }
    }

    return <a id={id} href={url} onClick={clickHandler} target="_blank" className='link-component'>{text}</a>
}

export default connect()(LinkComponent);