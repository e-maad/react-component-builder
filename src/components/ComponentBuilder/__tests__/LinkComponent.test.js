import React from 'react';
import LinkComponent from '../LinkComponent';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

const mockStore = configureStore([]);

describe('Link Component', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore({});
    wrapper = shallow(
      <Provider store={store}>
        <LinkComponent />
      </Provider>
    );
  });

  it('match the snapshot', () => {
    expect(renderer.create(<Provider store={store}>
      <LinkComponent />
    </Provider>).toJSON()).toMatchSnapshot();
  });

  it('base element should be present', () => {
    expect(wrapper.html().includes('link-component')).toBe(true);
  });
});