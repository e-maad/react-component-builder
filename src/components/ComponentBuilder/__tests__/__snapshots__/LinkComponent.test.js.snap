// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component match the snapshot 1`] = `
<a
  className="link-component"
  onClick={[Function]}
  target="_blank"
/>
`;
